File: src/main/java/com/hbx/play/daily/IsSubsequence392.java
package com.hbx.play.daily;

public class IsSubsequence392 {

    // 给定s，check s是否是t的子序列. s = "abc", t = "ahbgdc", -> true
    public static boolean isSubsequence(String s, String t) {
        if (s.length() <= 0 || s.equals(t)) {
            return true;
        }
        // 双指针来找t中符合s的位置
        int sLeft = 0, tLeft = 0;
        while (sLeft < s.length() && tLeft < t.length()) {
            if (s.charAt(sLeft) == t.charAt(tLeft)) {
                sLeft++;
            }
            tLeft++;
        }
        return sLeft == s.length();
    }

    public static void main(String[] args) {
        String s = "afc", t = "ahbgdc";
        System.out.println(isSubsequence(s, t));
    }
}

File: go/alg/is_subsequence_392.go
package main

import "fmt"

func main() {
	s := "abc"
	t := "ahbgdc"

	fmt.Println(isSubsequence(s, t))
}

func isSubsequence(s string, t string) bool {
	if len(s) <= 0 {
		return true
	}
	sLeft, tLeft := 0, 0

	for sLeft < len(s) && tLeft < len(t) {
		if s[sLeft] == t[tLeft] {
			sLeft++
		}
		tLeft++
	}
	return sLeft == len(s)
}
